mongo/change_stream.go

```diff
@@ -457,20 +457,33 @@ func (cs *ChangeStream) ResumeToken() bson.Raw {
 }
 
 // Next gets the next result from this change stream. Returns true if there were no errors and the next
-// result is available for decoding.
+// result is available for decoding. Next blocks until an event is available for decoding or ctx expires.
 func (cs *ChangeStream) Next(ctx context.Context) bool {
+	return cs.next(ctx, false)
+}
+
+// TryNext attempts to get the next result from this change stream. It returns true if there were no errors and the next
+// result is available for decoding. It returns false if the change stream was closed by the server, there was an
+// error getting more results from the server, or the server returned an empty batch of events. If an error occurred or
+// the stream was closed (can be checked with cs.ID() == 0), TryNext must not be called again.
```

Comment:
As above, I suggest noting the behavior of context expiration.  I'm also not clear why TryNext must not be called again?  Particularly as Next doesn't have a similar prohibition.  You might also consider enforcing such restriction, e.g. if `cs.Err() != nil` then return false up front in those functions.

Reply:
added this restriction